app/models/tag.rb

```diff
@@ -1311,6 +1328,7 @@ def after_update
       if tag.merger_id.present?
         tag.merger.update_works_index_timestamp!
       end
+      IndexQueue.enqueue_ids(Tag, tag.suggested_child_ids, :background)
```

Comment:
Damn. We do need to reindex all of the suggested child tags. We're trying to make it so even if I de-syn a big tag like Hockey RPF from the less frequently used canonical tag Men's Hockey RPF, all of the tags that had Hockey RPF as suggested parent will leave the unwrangled bins for Men's Hockey RPF and appear in the bins for Hockey RPF instead.

Reply:
I stuck this in a `queue_child_tags_for_reindex` method and then made it so the `after_update` calls `async(:queue_child_tags_for_reindex)` but after I did that, I realized I wasn't 100% sure that was what you meant. Do I need to make `suggested_child_ids` itself asynchronous? 

To get around the first page issue, I set the `per_page` on that query to `ArchiveConfig.MAX_SEARCH_RESULTS` results, but now I'm wondering if we should be going through the database instead to avoid having a limit. otoh, if there's ever a tag with over 100,000 suggested children, we probably don't want to be getting that from the database, either.

Reply:
Yes, that's what I meant in terms of making the computation asynchronous. Just calling `async(:queue_child_tags_for_reindex)` is fine.

I'm not sure whether it's reasonable to try to make ES return 100,000 records in one go, even if it is in a Resque job. It might be good to try a [scrolled search](https://www.elastic.co/guide/en/elasticsearch/reference/6.5/search-request-scroll.html), since it gets around the search results limit and loads results page by page:
```
query = suggested_child_tags_query

parameters = {
  index: query.index_name,
  type: query.document_type,
  body: { query: query.filtered_query, sort: "_doc", size: 1000, _source: false },
  scroll: "1m"
}

response = $elasticsearch.search(parameters)
while response.dig("hits", "hits").present?
  tag_ids = response.dig("hits", "hits").map { |hit| hit["_id"] }
  IndexQueue.enqueue_ids(Tag, tag_ids, :background)
  response = $elasticsearch.scroll(body: { scroll_id: response["_scroll_id"] }, scroll: "1m")
end

$elasticsearch.clear_scroll(scroll_id: response["_scroll_id"])
```
But this is complex enough that it might be worth rewriting as a general-purpose `scroll` function in the Query class that yields hits a little at a time. I don't like having this quantity of ES-related code outside of an ES-specific class.

Alternatively, if you'd prefer to work with the database instead of ES, you could load batches of works and queue up all the tags for each batch:
```
all_with_child_type = Tag.where(type: child_types & Tag::USER_DEFINED)
works.select(:id).find_in_batches do |batch|
  relevant_taggings = Tagging.where(taggable: batch)
  tag_ids = all_with_child_type.joins(:taggings).merge(relevant_taggings).distinct.pluck(:id)
  IndexQueue.enqueue_ids(Tag, tag_ids, :background)
end
```
I'm not sure which would be better/faster.

Reply:
We went with a database query, which is simpler for now.